Sources/StreamVideoSwiftUI/Utils/StreamPixelBufferRepository/StreamPixelBufferRepository.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CoreVideo
+import Foundation
+import StreamVideo
+
+/// A repository class that manages multiple pools of pixel buffers to efficiently handle various sizes and formats.
+final class StreamPixelBufferRepository {
+
+    /// A private struct to use as a key in the dictionary that maps buffer characteristics to their respective pools.
+    private struct Key: Hashable {
+        /// The width of the pixel buffers in the pool.
+        var width: Int
+        /// The height of the pixel buffers in the pool.
+        var height: Int
+        /// The pixel format type of the pixel buffers in the pool.
+        var pixelFormat: OSType
+
+        /// Initializes a new key with a given size and pixel format.
+        /// - Parameters:
+        ///   - size: The size of the pixel buffers.
+        ///   - pixelFormat: The pixel format type.
+        init(_ size: CGSize, pixelFormat: OSType) {
+            width = Int(size.width)
+            height = Int(size.height)
+            self.pixelFormat = pixelFormat
+        }
+    }
+
+    /// A dictionary that maps unique keys to specific `StreamPixelBufferPool` instances.
+    private var pools: [Key: StreamPixelBufferPool] = [:]
+
+    /// A dispatch queue used for synchronizing access to the pixel buffer pools.
+    private let queue = UnfairQueue()
```

Comment:
Do you think we could have priority inversion with this queue? Doesn't seem likely.

Reply:
I doubt it, because our requests to the queue are already serialized (from the way that frames are coming in)